ios/Mattermost/AppDelegate.m

```diff
@@ -71,24 +71,18 @@ -(void)application:(UIApplication *)application didReceiveRemoteNotification:(no
   NSString* action = [userInfo objectForKey:@"type"];
   NSString* channelId = [userInfo objectForKey:@"channel_id"];
   NSString* ackId = [userInfo objectForKey:@"ack_id"];
+  RuntimeUtils *utils = [[RuntimeUtils alloc] init];
 
-  if (action && [action isEqualToString: NOTIFICATION_CLEAR_ACTION]) {
+  if ((action && [action isEqualToString: NOTIFICATION_CLEAR_ACTION]) || (state == UIApplicationStateInactive)) {
     // If received a notification that a channel was read, remove all notifications from that channel (only with app in foreground/background)
     [self cleanNotificationsFromChannel:channelId];
```

Comment:
Is this safe with an empty channeId?

Reply:
This method does not get called for session expired notifications, only for clear notifications cause of what the payload contains